libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1873,6 +1873,92 @@ void GCS_MAVLINK::handle_vision_position_delta(mavlink_message_t *msg)
     visual_odom->handle_msg(msg);
 }
 
+void GCS_MAVLINK::handle_vision_position_estimate(mavlink_message_t *msg)
+{
+    mavlink_vision_position_estimate_t m;
+    mavlink_msg_vision_position_estimate_decode(msg, &m);
+
+    _handle_common_vision_position_estimate_data(m.usec, m.x, m.y, m.z, m.roll, m.pitch, m.yaw);
+}
+
+void GCS_MAVLINK::handle_global_vision_position_estimate(mavlink_message_t *msg)
+{
+    mavlink_global_vision_position_estimate_t m;
+    mavlink_msg_global_vision_position_estimate_decode(msg, &m);
+
+    _handle_common_vision_position_estimate_data(m.usec, m.x, m.y, m.z, m.roll, m.pitch, m.yaw);
+}
+
+void GCS_MAVLINK::handle_vicon_position_estimate(mavlink_message_t *msg)
+{
+    mavlink_vicon_position_estimate_t m;
+    mavlink_msg_vicon_position_estimate_decode(msg, &m);
+
+    _handle_common_vision_position_estimate_data(m.usec, m.x, m.y, m.z, m.roll, m.pitch, m.yaw);
+}
+
+// there are several messages which all have identical fields in them.
+// This function provides common handling for the data contained in
+// these packets
+void GCS_MAVLINK::_handle_common_vision_position_estimate_data(const uint64_t usec,
+                                                               const float x,
+                                                               const float y,
+                                                               const float z,
+                                                               const float roll,
+                                                               const float pitch,
+                                                               const float yaw)
+{
+    // sensor assumed to be at 0,0,0 body-frame; need parameters for this?
+    // or a new message 
+    const Vector3f sensor_offset = {};
```

Comment:
I was expecting this would go through the AP_VisualOdom library which already have those parameters.